src/SharedPipelineComponents/DataLockEvents/SFA.DAS.Provider.Events.DataLock/Application/GetCurrentProviderEvents/GetCurrentProviderEventsHandler.cs

```diff
@@ -83,14 +83,15 @@ public GetCurrentProviderEventsResponse Handle(GetCurrentProviderEventsRequest m
                         var collectionPeriod = GetCollectionPeriod(entity.Period);
                         if (!periods.Any(p => p.CollectionPeriod.Month == collectionPeriod.Month
                                            && p.CollectionPeriod.Year == collectionPeriod.Year
-                                           && (int)p.TransactionType == entity.TransactionType))
+                                           && (int)p.TransactionTypesFlag == entity.TransactionTypesFlag))
                         {
                             periods.Add(new DataLockEventPeriod
                             {
                                 CollectionPeriod = collectionPeriod,
                                 CommitmentVersion = entity.CommitmentVersionId,
                                 IsPayable = entity.Payable,
-                                TransactionType = (TransactionType)entity.TransactionType
+                                TransactionType = (TransactionType)entity.TransactionType,
+                                TransactionTypesFlag = (TransactionTypesFlag)entity.TransactionTypesFlag
                             });
                         }
 
```

Comment:
line 98 can be simplified to:

'if (entity.RuleId != null && errors.All(e => e.ErrorCode != entity.RuleId))

Line 107 can be simplified to:

'if (commitmentVersions.All(v => v.CommitmentVersion != entity.CommitmentVersionId))

Reply:
Consider refactoring GetCollectionPeriod method as the case statements are almost all identical and are just month/year shifting